README.md

```diff
@@ -1,39 +1,309 @@
+Material Text Field is a customizable widget for text input values in Dart. You can define the styling of the text field in your app's theme file or create multiple text fields with different styling.With this package, you can easily create text input fields with customizable styling and behaviors.
+## Installation
+To use Material Text Field in your Dart project, add the following dependency to your "pubspec.yaml" file
 
-For general information about developing packages, see the Dart guide for
-[creating packages](https://dart.dev/guides/libraries/create-library-packages)
-and the Flutter guide for
-[developing packages and plugins](https://flutter.dev/developing-packages).
--->
+  ```dart
+dependencies:
+  material_text_field: <latest-version>
+```
+Then run flutter pub get to install the package.
+
+In your library add the following import
 
-TODO: Put a short description of the package here that helps potential users
-know whether this package might be useful for them.
+```dart
+import 'package:material_text_fields/material_text_fields.dart';
+```
 
-## Features
+## Usage
+To use Material Text Field in your Flutter app, import the package and create a new instance of the MaterialTextField widget
 
-TODO: List what your package can do. Maybe include images, gifs, or videos.
+  ```dart
+             MaterialTextField(
+                keyboardType: TextInputType.emailAddress,
+                hint: "Email",
+                textInputAction: TextInputAction.next,
+                prefixIcon: Image.asset(ImagePath.icEmail),
+                controller: _emailController,
+                validator: FormValidation.emailTextField,
+              )
+```
 
-## Getting started
+You can create multiple text fields with different styling with providing "theme" attribute.
 
-TODO: List prerequisites and provide or point to information on how to
-start using the package.
+  ```dart
+              MaterialTextField(
+                keyboardType: TextInputType.text,
+                labelText: "Name",
+                theme: MaterialTextFieldTheme.filledOrOutlinedTextTheme(
+                  enabledColor: Colors.grey,
+                  focusedColor: Colors.green,
+                  fillColor: Colors.transparent,
+                ),
+                prefixIcon: Image.asset(ImagePath.icEmail),
+              )
+```
 
-## Usage
+## Advance Usage
+
+With all possible params
+
+  ```dart
+               MaterialTextField(
+                  keyboardType: TextInputType.emailAddress,
+                  hint: "Email",
+                  textInputAction: TextInputAction.next,
+                  prefixIcon: Image.asset(ImagePath.icLock),
+                  controller: _emailController,
+                  validator: FormValidation.emailTextField,
+                  onChanged: (text) {
+                    print('First text field: $text');
+                  },
+                  suffixIcon: Image.asset(ImagePath.icLock),
```

Comment:
use a hard-coded string value for the image path like `assets/images/ic_lock.png`